src/field/field2d.cxx

```diff
@@ -130,9 +130,8 @@ void Field2D::setLocation(CELL_LOC new_location) {
       new_location = CELL_CENTRE;
     }
 
-    fieldCoordinates = getCoordinates();
-
     location = new_location;
+    fieldCoordinates = getCoordinates();
```

Comment:
These calls could be just `getCoordinates()`. `fieldCoordinates` is already set inside `getCoordinates()`.

Reply:
Good point!